tidb-cloud/data-service-oas-with-nextjs.md

```diff
@@ -0,0 +1,209 @@
+                  "result": { ... },
+                  "rows": {
+                    "items": {
+                      "properties": {
+                        "id": {
+                          "type": "string"
+                        },
+                        "name": {
+                          "type": "string"
+                        },
+                        "url": {
+                          "type": "string"
+                        }
+    ...
+      "paths": {
+        "/repositories": {
+          "get": {
+            "operationId": "getRepositories",
+            "responses": {
+              "200": {
+                "content": {
+                  "application/json": {
+                    "schema": {
+                      "$ref": "#/components/schemas/getRepositoriesResponse"
+                    }
+                  }
+                },
+                "description": "OK"
+              },
+    ...
+    ```
+
+4. Generate the client code:
+
+    ```shell
+    yarn run openapi-generator-cli generate -i oas/doc.json --generator-name typescript-fetch -o gen/api
+    ```
+
+    This command generates the client code using the `oas/doc.json` specification as input and outputs the client code to the `gen/api` directory.
+
+## Step 4. Develop your Next.js application
+
+You can use the generated client code to develop your Next.js application.
+
+1. In the `hello-repos` project directory, create a `.env.local` file with the following variables, and then set the variable values to the public key and private key of your Data App.
+
+    ```
+    TIDBCLOUD_DATA_SERVICE_PUBLIC_KEY=YOUR_PUBLIC_KEY
+    TIDBCLOUD_DATA_SERVICE_PRIVATE_KEY=YOUR_PRIVATE_KEY
+    ```
+
+    To create an API key for a Data App, see [Create an API key](/tidb-cloud/data-service-api-key.md#create-an-api-key).
+
+2. In the `hello-repos` project directory, replace the content of `app/page.tsx` with the following code, which fetches data from the `GET /repositories` endpoint and renders it:
+
+    ```js
+    import {DefaultApi, Configuration} from "../gen/api"
+
+    export default async function Home() {
+      const config = new Configuration({
+        username: process.env.TIDBCLOUD_DATA_SERVICE_PUBLIC_KEY,
+        password: process.env.TIDBCLOUD_DATA_SERVICE_PRIVATE_KEY,
+      });
+      const apiClient = new DefaultApi(config);
+      const resp = await apiClient.getRepositories();
+      return (
+        <main className="flex min-h-screen flex-col items-center justify-between p-24">
+          <ul className="font-mono text-2xl">
+            {resp.data.rows.map((repo) => (
+              <a href={repo.url}>
+                <li key={repo.id}>{repo.name}</li>
+              </a>
+            ))}
+          </ul>
+        </main>
+      )
+    }
+    ```
+
+    > **Note:**
+    >
+    > If your Data App is deployed across multiple regions, you also need to configure the endpoint path in the `config` object as follows:
```

Comment:
```suggestion
    > If your Data App is deployed across multiple regions, which means it is linked to clusters in different regions, you will see multiple items in the `servers` section of the downloaded OpenAPI Specification file. You also need to configure the endpoint path in the `config` object as follows:
```

Reply:
If your Data App is deployed across multiple regions, which means its linked clusters are hosted in different regions, you will see multiple items in the `servers` section of the downloaded OpenAPI Specification file. In this case, you also need to configure the endpoint path in the `config` object as follows: